src/SignalR/clients/ts/signalr/src/HttpConnection.ts

```diff
@@ -353,15 +353,18 @@ export class HttpConnection implements IConnection {
         const transports = negotiateResponse.availableTransports || [];
         for (const endpoint of transports) {
             try {
-                const transport = this.resolveTransport(endpoint, requestedTransport, requestedTransferFormat);
-                if (typeof transport === "number") {
-                    this.transport = this.constructTransport(transport);
+                const transportOrError = this.resolveTransport(endpoint, requestedTransport, requestedTransferFormat);
```

Comment:
Should this be moved outside the try? It being the first statement in the try makes me think it can throw. If it cannot throw, moving it outside of the try makes the code easier to read/modify/review.